server/src/main/java/org/candlepin/resource/JobResource.java

```diff
@@ -174,6 +181,42 @@ public SchedulerStatus setSchedulerStatus(boolean running) {
         return getSchedulerStatus();
     }
 
+    @POST
+    @Path("cron/{task}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @SuppressWarnings("unchecked")
+    public JobStatus schedule(@PathParam("task") String task) {
+
+        String className = "org.candlepin.pinsetter.tasks." + task;
+        try {
+            boolean isCron = false;
+            for (String cronJob : ConfigProperties.DEFAULT_TASK_LIST) {
+                if (className.equalsIgnoreCase(cronJob)) {
+                    isCron = true;
+                    // helps ignore case
+                    className = cronJob;
+                }
+            }
+            if (isCron) {
+                Class<? extends KingpinJob> taskClass = (Class<? extends KingpinJob>) Class
+                    .forName(className);
+                return pk.scheduleSingleJob(taskClass, Util.generateUUID());
```

Comment:
ClassCastException in either case, one line away from the declaration. You never actually use the class outside of this line, so if you're going to start line wrapping for the sake of exception accuracy, you may as well drop the extraneous declaration and do it all in one line:

```
return pk.scheduleSingleJob((Class<? extends KingpinJob>) Class.forName(className),
    Util.generateUUID());
```